rest-api-spec/src/main/resources/rest-api-spec/api/synonyms.put.json

```diff
@@ -0,0 +1,39 @@
+{
+  "synonyms.put": {
+    "documentation": {
+      "url": "https://www.elastic.co/guide/en/elasticsearch/reference/master/put-synonyms.html",
+      "description": "Creates or updates a synonym set"
```

Comment:
Nit: I think we should be consistent everywhere (including API): either "synonym set" or "synonyms set". I am ok with either version (including API).

Reply:
Agreed, changing all instances I found in the code 👍